src/__init__.py

```diff
@@ -238,51 +232,41 @@ def generate_basic(self):
-
-            if "dont_place_item" in manual_location:
-                if len(manual_location["dont_place_item"]) == 0:
-                    continue
-
-                eligible_items = [item for item in eligible_items if item.name not in manual_location["dont_place_item"]]
-
-                if len(eligible_items) == 0:
-                    raise Exception("Could not find a suitable item to place at %s. No items that match placed_items(_category) because of forbidden %s." % (manual_location["name"], ", ".join(manual_location["dont_place_item"])))
+            #First we get possible items names
+            if manual_location.get("place_item"):
+                eligible_item_names += manual_location["place_item"]
+                place_messages.append('", "'.join(manual_location["place_item"]))
 
-            if "dont_place_item_category" in manual_location:
-                if len(manual_location["dont_place_item_category"]) == 0:
-                    continue
+            if manual_location.get("place_item_category"):
+                eligible_item_names += [i["name"] for i in item_name_to_item.values() if "category" in i and set(i["category"]).intersection(manual_location["place_item_category"])]
+                place_messages.append('", "'.join(manual_location["place_item_category"]) + " category(ies)")
 
-                forbidden_item_names = [i["name"] for i in item_name_to_item.values() if "category" in i and set(i["category"]).intersection(manual_location["dont_place_item_category"])]
+            # Second we check for forbidden items names
+            if manual_location.get("dont_place_item"):
+                forbidden_item_names += manual_location["dont_place_item"]
+                forbid_messages.append('", "'.join(manual_location["dont_place_item"]) + ' items')
 
-                eligible_items = [item for item in eligible_items if item.name not in forbidden_item_names]
+            if manual_location.get("dont_place_item_category"):
+                forbidden_item_names += [i["name"] for i in item_name_to_item.values() if "category" in i and set(i["category"]).intersection(manual_location["dont_place_item_category"])]
+                forbid_messages.append('", "'.join(manual_location["dont_place_item_category"]) + ' category(ies)')
 
-                if len(eligible_items) == 0:
-                    raise Exception("Could not find a suitable item to place at %s. No items that match placed_items(_category) because of forbidden categories %s." % (manual_location["name"], ", ".join(manual_location["dont_place_item_category"])))
-                forbidden_item_names.clear()
+            # If we forbid some names, check for those in the possible names and remove them
+            if forbidden_item_names:
+                eligible_item_names = [name for name in eligible_item_names if name not in forbidden_item_names]
```

Comment:
actually this entire loop is only entered if theres either a place_item or place_item_category
https://github.com/ManualForArchipelago/Manual/blob/place_item_rewrite/src/__init__.py#L230-L232
I previously used it as a forbid placement when there's no place_item(_cat) so that might be a problem to add it like you propose

Reply:
Oh, I see now. I missed that there was a separate set of handling for dont_place and dont_place_cat above this. This is just removing from the place eligible list, makes sense. Nevermind 😄